### DISCUSSION TITLE
Intellisense for NeoVim libraries in Lua

### DISCUSSION THREAD

**Discussion by ollien** [vscode-neovim/vscode-neovim#2074] [Development]
I haven't found a great way to do this, and maybe other maintainers have already figured it out. Is there a way to get completions for the neovim API when writing Lua scripts? This project has been my first toes dipped in Lua, so I'm not hugely familiar with the ecosystem, and my searches have come up empty. If there is, I'm happy to write it up formally for `CONTRIBUTING.md` :)

Any pointers would be appreciated!

**Comment by ollien** [comment]
After some hunting, I think I figured it out! I've been using [this extension](https://marketplace.visualstudio.com/items?itemName=sumneko.lua), and I added the following to my configuration


```json
		"Lua.workspace.library": [
			"/usr/share/nvim/runtime/lua/vim"
		],
```

I don't think we can reasonably commit this to `.vscode/settings.json`, because the path will be different for everyone, but I can do some investigative work to figure out the path for various platforms, and write something more formal up.

**Comment by theol0403** [comment]
https://github.com/vscode-neovim/vscode-neovim/blob/master/CONTRIBUTING.md#set-up-vscode-for-lua-development

**Reply by ollien** [reply]
:expressionless: Can't believe I missed that this whole time
